api/middleware/auth.validation.js

```diff
@@ -0,0 +1,23 @@
+const Joi = require('joi')
+
+const validate = {
+   
+   login:(username, password) =>{
+       const Schema = Joi.object({
+           username: Joi.string().min(6).required(),
+           password: Joi.string().min(5)
+       })
+return Schema.validate({username, password})
+   },
+   signup:(firstname, lastname ,username ,email , password) =>{
+       const Schema = Joi.object({
+           firstname: Joi.string().min(3).required(),
+           lastname: Joi.string().min(3).required(),
+           username: Joi.string().min(3).required(),
```

Comment:
You are checking for different things for username in signup and login, with respect to the minimum length